crypto/evp/evp_enc.c

```diff
@@ -87,6 +89,9 @@ static int evp_cipher_init_internal(EVP_CIPHER_CTX *ctx,
 #if !defined(OPENSSL_NO_ENGINE) && !defined(FIPS_MODULE)
     ENGINE *tmpimpl = NULL;
 #endif
+
+    ctx->iv_len = -1;
```

Comment:
Shouldn't this only happen on `iv != NULL`?

Reply:
I don't feel that such an optimisation is all that relevant.  This is just flagging that the cache needs to be reloaded, better to do this conservatively than not.

Can we guarantee that a cipher will never change its IV length when the key is set or vice versa?